src/NuGetGallery.Core/GitHub/RepositoryInformation.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace NuGetGallery.GitHub
+{
+    public class RepositoryInformation
+    {
+        public RepositoryInformation(
+            string id,
+            string url,
+            int stars,
+            IReadOnlyList<string> dependencies)
+        {
+            if (stars < 0)
+            {
+                throw new IndexOutOfRangeException(string.Format("{0} cannot have a negative value!", nameof(stars)));
+            }
+
+            Id = id;
+            var idSplit = Id.Split('/');
+            if (idSplit.Length == 2)
+            {
+                Owner = idSplit[0];
+                Name = idSplit[1];
+            }
+            else
+            {
+                throw new ArgumentException(string.Format("{0} has an invalid format! It should be \"owner/repositoryName\", instead it is: {1}", nameof(Id), Id));
```

Comment:
We shouldn't log this as the "owner" is personally identifying information (PII)